synapseclient/models/protocols/json_schema_protocol.py

```diff
@@ -624,14 +544,94 @@ def get_invalid_validation(
+
+        Arguments:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Yields:
+            InvalidJSONSchemaValidation: An object containing the validation response, all validation messages,
+                                         and the validation exception details.
+
+        Example: Using this function
+            Retrieving invalid validation results for a folder
+
+        ```python
+        import synapseclient
+        from synapseclient.models import File, Folder
+
+        syn = synapseclient.Synapse()
+        syn.login()
+
+        # Define Project and JSON schema info
+        PROJECT_ID = syn.findEntityId(name="test_json_schema_project")  # Replace with your project name
+        ORG_NAME = "UniqueOrg"  # Replace with your organization name
+        SCHEMA_NAME = "myTestSchema"  # Replace with your schema name
+        VERSION = "0.0.1"
+        SCHEMA_URI = f"{ORG_NAME}-{SCHEMA_NAME}-{VERSION}"
+
+        # Create organization (if not already created)
+        js = syn.service("json_schema")
+        all_orgs = js.list_organizations()
+        for org in all_orgs:
+            if org["name"] == ORG_NAME:
+                print(f"Organization {ORG_NAME} already exists.")
+                break
+        else:
+            print(f"Creating organization {ORG_NAME}.")
+            js.create_organization(ORG_NAME)
+
+        # Create the schema (if not already created)
+        schema_definition = {
+            "$id": "mySchema",
+            "type": "object",
+            "properties": {
+                "foo": {"type": "string"},
+                "baz": {"type": "string", "const": "example_value"},  # Example constant for derived annotation
+                "bar": {"type": "integer"},
+            },
+            "required": ["foo"]
+        }
+
+        my_test_org = js.JsonSchemaOrganization(ORG_NAME)
+        test_schema = my_test_org.get_json_schema(SCHEMA_NAME)
+        if not test_schema:
+            test_schema = my_test_org.create_json_schema(schema_definition, SCHEMA_NAME, VERSION)
+
+        # Create a test folder
+        test_folder = Folder(name="test_script_folder", parent_id=PROJECT_ID)
+        test_folder.store()
+
+        # Bind JSON schema to the folder
+        bound_schema = test_folder.bind_schema(
+            json_schema_uri=SCHEMA_URI,
+            enable_derived_annotations=True
+        )
+
+        # Validate the folder entity against the bound schema with invalid data
+        test_folder.annotations = {"foo": "test_value_new", "bar": "invalid_string"}  # 'bar' should be integer
+        test_folder.store()
+
+        # Get invalid validation results
+        invalid_items = test_folder.get_invalid_validation()
+        for item in invalid_items:
+            print(item)
```

Comment:
@linglp This example doesn't seem to actually work. Am I doing something wrong, or reading the API docs incorrectly?

My understanding after reading https://rest-docs.synapse.org/rest/POST/entity/id/schema/validation/invalid.html is that it won't validate the Folder itself, but rather, the content within the folder.

Reply:
Your understanding is correct. That's my mistake. Let me run and fix all these examples